src/main/scala/fr/acinq/bitcoin/Script.scala

```diff
@@ -325,11 +325,9 @@ object Script {
     // We want to compare apples to apples, so fail the script
     // unless the type of nLockTime being tested is the same as
     // the nLockTime in the transaction.
-    if (!(
-      (tx.lockTime < Transaction.LOCKTIME_THRESHOLD && lockTime < Transaction.LOCKTIME_THRESHOLD) ||
-        (tx.lockTime >= Transaction.LOCKTIME_THRESHOLD && lockTime >= Transaction.LOCKTIME_THRESHOLD)
-      ))
+    if (!(tx.lockTime < Transaction.LOCKTIME_THRESHOLD && lockTime < Transaction.LOCKTIME_THRESHOLD) && !(tx.lockTime >= Transaction.LOCKTIME_THRESHOLD && lockTime >= Transaction.LOCKTIME_THRESHOLD)) {
```

Comment:
Nit: our check matches exactly what bitcoin core is doing, now it will look different

Reply:
Ok, I can revert then, in my opinion `!A && !B` is clearer than `!(A || B)` but it's better to be consistent with bitcoin core